Flick/Views/LoginButton.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  LoginButton.swift
+//  Flick
+//
+//  Created by Haiying W on 1/14/21.
+//  Copyright © 2021 flick. All rights reserved.
+//
+
+import UIKit
+
+enum LoginButtonType: String {
+    case apple = "Apple"
+    case facebook = "Facebook"
+}
+
+class LoginButton: UIButton {
+
+    private let iconImageView = UIImageView()
+    private let label = UILabel()
+
+    init(type: LoginButtonType, backgroundColor: UIColor) {
+        super.init(frame: .zero)
+        self.backgroundColor = backgroundColor
+        clipsToBounds = true
+        layer.cornerRadius = 23
+
+        var iconImageName: String {
```

Comment:
technically if we already have the enum and you specify a string for each type, can't you just directly use the enum?